probatus/binning/binning.py

```diff
@@ -437,9 +436,8 @@ def tree_bins(x, y, inf_edges, tree):
 
         bin_count = len(index)
 
-        tpf = TreePathFinder(tree)
-        boundaries = [bound["min"] for bound in tpf.get_boundaries().values()]
-        boundaries += [tpf.get_boundaries()[leaves[-1]]["max"]]
+        boundaries = np.unique(tree.tree_.threshold[tree.tree_.feature != _tree.TREE_UNDEFINED])
+        boundaries = [-np.inf] + boundaries.tolist() + [np.inf]
 
         if not inf_edges:
```

Comment:
Maybe to be fully consistent with other classes, in line 440 add as boundaries min and max for the data, and then 

```
if inf_edge:
    change the boundaries to infinite
```

Reply:
done.